orchagent/flexcounterorch.cpp

```diff
@@ -0,0 +1,76 @@
+#include <unordered_map>
+#include "flexcounterorch.h"
+#include "portsorch.h"
+#include "select.h"
+#include "notifier.h"
+#include "redisclient.h"
+#include "sai_serialize.h"
+
+extern sai_port_api_t *sai_port_api;
+
+extern PortsOrch *gPortsOrch;
+
+unordered_map<string, string> flexCounterGroupMap =
+{
+    {"PORT", PORT_STAT_COUNTER_FLEX_COUNTER_GROUP},
+    {"QUEUE", QUEUE_STAT_COUNTER_FLEX_COUNTER_GROUP},
+};
+
+FlexCounterOrch::FlexCounterOrch(DBConnector *db, vector<string> &tableNames):
+    Orch(db, tableNames),
+    m_flexCounterDb(new DBConnector(FLEX_COUNTER_DB, DBConnector::DEFAULT_UNIXSOCKET, 0)),
+    m_flexCounterGroupTable(new ProducerTable(m_flexCounterDb.get(), FLEX_COUNTER_GROUP_TABLE))
+{
+    SWSS_LOG_ENTER();
+}
+
+FlexCounterOrch::~FlexCounterOrch(void)
+{
+    SWSS_LOG_ENTER();
+}
+
+void FlexCounterOrch::doTask(Consumer &consumer)
+{
+    SWSS_LOG_ENTER();
+
+    if (!gPortsOrch->isInitDone())
+    {
+        return;
+    }
+
+    auto it = consumer.m_toSync.begin();
+    while (it != consumer.m_toSync.end())
+    {
+        KeyOpFieldsValuesTuple t = it->second;
+
+        string key =  kfvKey(t);
+        string op = kfvOp(t);
+        auto data = kfvFieldsValues(t);
+
+        if (!flexCounterGroupMap.count(key))
+        {
+            SWSS_LOG_NOTICE("Invalid flex counter group input, %s", key.c_str());
+            consumer.m_toSync.erase(it++);
+            continue;
+        }
+
+        if (op == SET_COMMAND)
```

Comment:
IFAIK if nothing is set, the default value of 1 second is used.
Then I think we also have to set the default back if the entry is deleted from DB.

Reply:
This entry only has set_command and no del command. We can always use set command to change the interval value.

Reply:
ok